### ISSUE TITLE
Use case: jwt auth

### ISSUE THREAD

**Issue by pumpk1n** [angular/router#253]
Hey!

I'm creating a couple of new sites and all of them need an auth system. I've used jwt token based auth before with ui-router and it all worked fine, but I decided I wanted to give a shot to the new router. My first thought was why not try with angular 2.0, but since there's no example and it's still in the early alpha stage I dropped the idea and stayed with 1.4.x.

Then I realized there isn't any example with an auth system either. I already read #33 and #170, but I still don't quite understand what would be the best practice to create a jwt token based auth system with the new router. It would be nice to have something like this, but it might be a long shot:

	$router.config([
		{ path: '/'			, redirectTo: 'index' 	, auth: false },
		{ path: '/profile'	, component: 'profile' 	, auth: true },
		{ path: '/login'	, component: 'login' 	, auth: false },
		...
	]);

And then redirect the user wherever if auth fails.

Could someone please explain how I could implement this step by step?

Thanks!

**Comment by tommck** [comment]
I'm guessing you'd want to do something with the canActivate lifecycle event to check for your auth situation

**Comment by pumpk1n** [comment]
Yeah I saw that in the documentation, I was just hoping in something with a little bit more details. Anyways, I just noticed the latest versions don't really work (`path undefined`), so I'm kinda stuck.
